### DISCUSSION TITLE
Simple example of Dev Tools with Selenium 4

### DISCUSSION THREAD

**Discussion by MinxOfInsolence** [webdriverio/webdriverio#7976] [Q&A]
Hi there,

I am currently trying to experiment with WDIO and want to use the Dev tools service that is available with Selenium 4.

I am very very new to WDIO and new-ish to Node JS (coming from a Java background) and I have tried a number of examples that are out there and all seem to have errors ranging from non descript ones to the most common one which is that the "createCDPConnection" method is not found. I know it must be something simple I am missing.

So I'm wondering if someone has a simple working example of a WDIO project using Selenium 4 and the dev tools contained within it?

Would really appreciate any help anyone can provide because I am keen to learn more.

Thanks!

**Comment by portyu9** [comment]
Hi @MinxOfInsolence 

`createCDPConnection()` is not a WebdriverIO browser command. That API comes from Selenium language bindings/examples, which is why copying those examples into WDIO gives you "method is not found".

With WebdriverIO you normally create a standard WebDriver session and use WebdriverIO's APIs.

For a Chromium-specific CDP command, WDIO exposes the Chromium protocol command directly, for example:

```js
await browser.sendCommand('Network.enable', {})
```

For higher-level browser automation you should prefer normal WebdriverIO APIs (and, on current WDIO/browser versions, WebDriver BiDi) rather than building the test around Selenium Java's DevTools API.

So you don't need a Selenium-Java-style `createCDPConnection()` setup in a WDIO project. Use the WebdriverIO command/API corresponding to the DevTools feature you actually need.

If you can specify which DevTools feature you're trying to use — network interception, console events, performance metrics, etc. — that determines the exact WDIO API.

I reviewed the complete discussion; there were no replies, screenshots, attachments, or reproduction links.

Please mark this answer as accepted if it helped, thank you.
